### DISCUSSION TITLE
Question: how to detect relationships

### DISCUSSION THREAD

**Discussion by jorismak** [wopian/kitsu#846] [Question / Answer] [answered]
When data is requested , attributes and relationships are merged into one data object it seems. Which is nice.

But i need a list of relationships to know which subrequests to make.

For example , a field may contain an array of relationships. I don't know which those will be ahead of time. So after the first request , i have to make a separate call to fetch all relationships . But how can j reliably tell if a property was original a relationship or not ? Just assuming it was a relationship when an 'id' and 'type' property are present feels a bit dirty.

**Comment by wopian** [comment]
Sorry, I'm not really understanding what this question is asking. 

Resource linkages are defined in https://jsonapi.org/format/#document-resource-object-linkage (an object or array of objects with `id` and `type` strings)

**Comment by jorismak** [comment]
Since the 'relationships' key from the output document is removed , how can
i differentiate between what was a relationship and what is a normal
attribute value ?

**Comment by wopian** [comment, marked as answer]
Relationships will always be an object containing a `data` attribute (which itself will always contain `id`, `type` and any attributes from the relationship. 1-1 relationships will have `data` as an object and 1-many relationships will have `data` as an array of objects

I.e `author` and `comments` are 1-1 and 1-many relationships here:

```json5
{
  id: '89',
  type: 'post',
  content: 'My post content',
  author: {
    data: {
      id: '42',
      type: 'people',
      name: 'John'
    }
  },
  comments: {
    data: [
      {
        id: '43',
        type: 'comment',
        text: 'A comment',
        author: {
          data: {
            id: '3',
            type: 'people',
            name: 'Bob'
          }
        }
      }
    ]
  }
}
```

**Comment by jorismak** [comment]
Oh, stupid me :(. Missed that data wasn't flattened . That'll be specific
enough . Thanks !
